README.md

```diff
@@ -1,33 +1,164 @@
-If you're interested in automated dependency updates make sure that `renovate` has access to the new repository.
-
-# Things to do change when you use this repository as a template:
-- [ ] replace `example-pkg` with the name of your package in this `README`
-- [ ] update the `@smg-automotive/example` with the name of your package in `package.json`
-- [ ] update `repository` and `homepage` sections in `package.json` to point to your GitHub repository
-- [ ] ensure that Frontend team has the admin access to the repository
-- [ ] ensure that the Bots team has write access to the repository (this is needed to release the package)
-- [ ] provide usage examples in the `README.md`
-- [ ] enable the project on circleci.com to build and test your package
-- [ ] change the circleci status badge in this `README` to the new project
-- [ ] develop an awesome package
-- [ ] live long and prosper
-# example-pkg
-
-[![CircleCI](https://circleci.com/gh/smg-automotive/example-pkg/tree/main.svg?style=svg&circle-token=c183f151fea3c74453cf8dd962d31e115906a300)](https://circleci.com/gh/smg-automotive/example-pkg/tree/main)
+# api-client-pkg
+
+[![CircleCI](https://circleci.com/gh/smg-automotive/api-client-pkg/tree/main.svg?style=svg&circle-token=c183f151fea3c74453cf8dd962d31e115906a300)](https://circleci.com/gh/smg-automotive/example-pkg/tree/main)
 [![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
 
 ## Usage
+
 ```
-npm install @smg-automotive/example
+npm install @smg-automotive/api-client-pkg
 ```
 
+````typescript
+import { ApiClient } from "@smg-automotive/api-client-pkg"
+````
+
+### Configuration
+
+The available instance methods are listed below. The specified config will be merged with the instance config.
+
+````typescript
+ApiClient.configure({
+  baseUrl: 'https://api.automotive.ch/api',
+  headers: { 'Content-Type': 'application/json' },
+})
+````
+
+### Instance methods
+
+The ApiClient provides promise-based methods for `GET`, `POST`, `PUT` and `DELETE`.
+
+### Typing
+
+You can define the request and the response type of your method as follows:
+
+````typescript
+interface Car {
+  make: string
+  horsepower: number
+}
+
+// data has type Car
+var data = await ApiClient.get<Car>('/listings/search')
+
+// data has type { id: string } and body is typed as Car
+var data = await ApiClient.post<{ id: string }, Car>('/listings/create',
+  { make: 'bmw', horsepower: 300 },)
+````
+
+### Parameters
+
+If your request URL contains parameters, you can use curly brackets and the ApiClient replaces it with the passed data.
+This is aligned with the OpenAPI specification used by Swagger. Hence, you can easily search for occurrences of your
+APIs.
+
+````typescript
+// fetch is called with https://baseUrl.api.ch/dealers/123/listings/456
+await ApiClient.get('dealers/{dealerId}/listings/{listingId}', {
+  params: { dealerId: 123, listingId: 456 },
```

Comment:
regarding the comment above - body is a very similar concept to params, so I'd align